frontend/src/context/ArticleContext.js

```diff
@@ -10,9 +10,10 @@ export const articlesReducer = (state, action) => {
       };
 
     case "ADD":
-      return {
-        articles: [action.payload, ...state.articles],
-      };
+      if (state.articles == null) {
+        return {articles: [action.payload], }; //cannot iterate over null
+      }
+      return {articles: [action.payload, ...state.articles], };
```

Comment:
Perhaps you could do something like
```js
return state.articles ? {articles: [action.payload, ...state.articles], } : {articles: [action.payload], };
```
or something using the [`??` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)

Maybe it would sacrifice too much the code clearness :confused: 

At least we could use a `===` instead of  `==`
But I'm not quite sure so once again @atxr :smiling_face_with_tear: ?

Reply:
My bad I actually didn't quite understand the real problem, see my other thread on `dispatch` call!

Reply:
A real *Never Nester* would opt for
```js
state.articles ??= [] 
return {articles: [action.payload, ...state.articles], };
```

Reply:
Not exactly enlightened in `JS`, so I was unaware of all the equality stuff, but I do now 👍 fixed in 3b178585584f276ceabc0c56b954f8a59510594d

Not going to used any `??` shenanigan tho because I think it's not necessary, the nesting is not excessive I believe, so let's keep it clear 😉

Reply:
Perfect good for me!

Reply:
I'll let @ctmbl resolve

Reply:
> A real Never Nester would opt for

Ow god that was exactly what I was looking for but didn't manage to find it :heart_eyes: